components/button/button.tsx

```diff
@@ -0,0 +1,21 @@
+import { ButtonHTMLAttributes, FC } from 'react';
+import Link from 'next/link';
+
+export enum BUTTON_COLORS {
+  Green = 'green',
+  Purple = 'purple',
+  White = 'white',
+  Pink = 'pink',
+  Red = 'red',
+  Transparent = 'transparent',
+}
+
+export const Button = ({ children, className = '', buttonColor, href }: ButtonProps) => {
+  return (
+    <Link href={href}>
+      <a className={`className buttonColor`}>{children}</a>
+    </Link>
+  );
+};
```

Comment:
This `className` prop is not actually styling the button. In order to actually apply some styles, the value passed to `style` must be a valid CSS string, like `color: red`. Or we need to have a style defined in a stylesheet  somewhere in the app that matches the name passed to `className`, [like this](https://reactjs.org/docs/faq-styling.html#what-is-css-in-js)

This is actually generating an `a` element where the class is `className buttonColor`. 

Do we want to use the `className` to style this button? I'm not sure what's the purpose of the `buttonColor` prop then if we want to style this using the `className`? Or are you planning to have stylesheets so that we can use `className`? Btw, this applies to everywhere you used `className`

Also, seems like you wanted to use string interpolation to pass the actual values of the props into the `className`, but you forgot to add the `${}` surrounding the variable name.

Reply:
I am using classic class names now but I will probably change this once I add the styling later. I meant to use string interpolation here but forgot the `${}`.